File: SpringTILCode/src/main/java/com/example/springtilcode/domain/post/entity/Post.java
package com.example.springtilcode.domain.post.entity;

import com.example.springtilcode.domain.user.entity.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String title;

    @Column
    private String content;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;
}

File: kotlin/src/main/kotlin/com/example/demo/domain/post/entity/Post.kt
package com.example.demo.domain.post.entity

import com.example.demo.domain.user.entity.User
import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

@Entity
data class Post (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?= null,

    @Column
    var title: String?= null,

    @Column
    var content: String?= null,

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "user_id")
    var user: User?= null
){
    fun editPost(title: String?, content: String?){
        this.title = title
        this.content = content
    }
}